### DISCUSSION TITLE
Problems running the SofaRos example

### DISCUSSION THREAD

**Discussion by ToriWalnuss** [sofa-framework/sofa#2695] [Build / Config / Environment] [answered]
Hey everyone!

I am trying to run the Sofa ROS example, using version 20.12.01 downloaded from the SOFA website on Ubuntu Ubuntu 20.04.3 LTS, as explained here
https://github.com/SofaDefrost/SoftRobots/tree/d063e9cbbd883b79210b6cece706916faa409997/docs/examples/sofaros

However I am getting the following error and I do not know how to resolve it:
![image](https://user-images.githubusercontent.com/94621355/154105885-24cf06a3-7fe0-40d1-bd0b-83a7390a9e39.png)

Am I missing something?
Any help is highly appreciated! Let me know if you need more information.

Best regards and many thanks in advance,

Tori

**Comment by damienmarchal** [comment, marked as answer]
Hi @ToriWalnuss 

Thank you for reporting this issue. 

It has been a long time I didn't looked at this example. 
The example we not updated since SOFA switched from python2 to python3, so it is not surprising it is not working anymore. 

@samFabrice could you  update the example  (as you probably have to do sofa+ros very soon) ? 

 EDIT: 
 @ToriWalnuss if you are willing to fix it yourself, they way to go is to learn how the SofaPython2's PythonScriptController was transformed into a SofaPython3's Sofa.Core.Controller. There are examples for python3 controller in Sofapython3's plugin

**Reply by ToriWalnuss** [reply]
Hi @damienmarchal 

Thank you so much for your quick response!

I am not very familiar with python yet (just started learning it ;)) but I am very happy for your hint and will definitely look into that!
(However I wouldn't bet on me solving it as a newbie;))

I am more than happy to help and contribute if I can and will keep you in the loop as I guess quite a few people could be interested in a working example for SofaRos!

Best regards!

**Reply by damienmarchal** [reply]
Hi, 

I see that @EulalieCoevoet answered to your question in https://github.com/SofaDefrost/SoftRobots/discussions/105
So let's continue the discussion there.

**Comment by samFabrice** [comment]
Hi @damienmarchal , @ToriWalnuss 
ok I will correct that since I will have to use it in my project

**Reply by ToriWalnuss** [reply]
Hey @samFabrice,

that is really cool! I will also use it in my project so I am very happy for your quick responses and your help!
